srml/support/src/traits.rs

```diff
@@ -648,3 +648,24 @@ pub trait ChangeMembers<AccountId> {
 impl<T> ChangeMembers<T> for () {
 	fn change_members(_incoming: &[T], _outgoing: &[T], _new_set: &[T]) {}
 }
+
+/// A generic trait for reporting slashing violations.
+pub trait ReportSlash<Misbehaved, Reporter, Hash> {
+	/// Reports slashing for a misconduct where `to_slash` is the misbehaved entities and
+	/// `to_reward` is the entities that detected and reported the misbehavior
+	///
+	/// Returns `Ok` if the misconduct was unique otherwise `Err`
+	fn slash(to_slash: Misbehaved, to_reward: Reporter, footprint: Hash) -> Result<(), ()>;
```

Comment:
`Reporter` -> `Reporters`. Maybe we should use `Err` in case the function fail (but the misbehavior was unique) and return `false` if the misbehavior is duplicated.